core/shared/src/main/scala/io.buildo.metarpheus/core/Metarpheus.scala

```diff
@@ -2,12 +2,26 @@ package io.buildo.metarpheus
 package core
 
 import scala.meta._
-import scala.meta.inputs.Input
+import scala.meta.inputs.Input.File
+import scala.meta.io.AbsolutePath
+import scala.meta.internal.io.PlatformFileIO
 
 object Metarpheus {
 
-  def run(files: List[String], config: Config): intermediate.API = {
-    val parsed = files.map(Input.String(_).parse[Source].get)
+  def recursivelyListFiles(target: AbsolutePath): List[AbsolutePath] = {
+    if (target.isDirectory) {
+      val dirContent = PlatformFileIO.listFiles(target).iterator.toList
+      dirContent ++ dirContent.filter(_.isDirectory).flatMap(recursivelyListFiles)
```

Comment:
good point, I just copied this over from the existing implementation, and it seems to be working for now, but I'll double check.